service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/StructuredRecordMappingService.java

```diff
@@ -0,0 +1,61 @@
+package uk.nhs.adaptors.gp2gp.gpc;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.DocumentReference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.EhrExtractMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.MessageContext;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.OutputMessageWrapperMapper;
+import uk.nhs.adaptors.gp2gp.ehr.utils.DocumentReferenceUtils;
+import uk.nhs.adaptors.gp2gp.ehr.utils.ResourceExtractor;
+import uk.nhs.adaptors.gp2gp.mhs.model.OutboundMessage;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Service
+public class StructuredRecordMappingService {
+    private final OutputMessageWrapperMapper outputMessageWrapperMapper;
+    private final EhrExtractMapper ehrExtractMapper;
+    private final MessageContext messageContext;
+
+    public List<OutboundMessage.ExternalAttachment> getExternalAttachments(Bundle bundle) {
+        return ResourceExtractor.extractResourcesByType(bundle, DocumentReference.class)
+            .map(this::buildExternalAttachment)
+            .collect(Collectors.toList());
+    }
+
+    private OutboundMessage.ExternalAttachment buildExternalAttachment(DocumentReference documentReference) {
+        var attachment = DocumentReferenceUtils.extractAttachment(documentReference);
+        var referenceId = messageContext.getIdMapper().get(ResourceType.DocumentReference, documentReference.getId());
+        var manifestReferenceId = messageContext.getIdMapper().getOrNew(ResourceType.DocumentManifest, documentReference.getId());
+
+        return OutboundMessage.ExternalAttachment.builder()
+            .referenceId(referenceId)
+            .hrefId(manifestReferenceId)
+            .contentType(DocumentReferenceUtils.extractContentType(attachment))
+            .filename(DocumentReferenceUtils.buildAttachmentFileName(referenceId, attachment))
+            .length(attachment.getSize())
+            .compressed(false) //TODO: has GP2GP compressed this file https://gpitbjss.atlassian.net/browse/NIAD-1274
```

Comment:
We've backlogged 1274 since compression is not required. GPC will not compress the binary content. FHIR requires the binary content to match the contentType.